src/matcher_combinators/matchers.clj

```diff
@@ -1,31 +1,39 @@
 (ns matcher-combinators.matchers
   (:require [matcher-combinators.core :as core]))
 
-(defn equals-value
+(defn equals
   "Matcher that will match when the given value is exactly the same as the
   `expected`."
   [expected]
-  (core/->Value expected))
+  (cond
+    (sequential? expected) (core/->EqualsSeq expected)
+    (set? expected)        (core/->EqualsSet expected false)
+    (map? expected)        (core/->EqualsMap expected)
+    :else                  (core/->Value expected)))
+
+(defn equals-set
+  "Matches a set in the way `(equals some-set)` would, but accepts sequences as
+  the expected matcher argument, allowing one to use matchers with the same
+  submatcher appearing more than once."
+  [expected]
+  (core/->EqualsSet expected true))
 
-(defn contains-map
+(defn embeds
```

Comment:
I chose `embeds` because it won't conflict with midje's `contains` when people do `[midje.sweet :refer :all]`